cmd/kubeadm/app/cmd/token.go

```diff
@@ -208,46 +204,30 @@ func NewCmdTokenGenerate(out io.Writer) *cobra.Command {
 }
 
 // RunCreateToken generates a new bootstrap token and stores it as a secret on the server.
-func RunCreateToken(out io.Writer, client clientset.Interface, cfgPath string, initCfg *kubeadmapiv1beta2.InitConfiguration, printJoinCommand bool, kubeConfigFile string) error {
-	// ClusterConfiguration is needed just for the call to LoadOrDefaultInitConfiguration
-	clusterCfg := &kubeadmapiv1beta2.ClusterConfiguration{
-		// KubernetesVersion is not used, but we set this explicitly to avoid
-		// the lookup of the version from the internet when executing LoadOrDefaultInitConfiguration
-		KubernetesVersion: kubeadmconstants.CurrentKubernetesVersion.String(),
-	}
-	kubeadmscheme.Scheme.Default(clusterCfg)
-
-	// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
-	klog.V(1).Infoln("[token] loading configurations")
-
-	// In fact, we don't do any CRI ops at all.
-	// This is just to force skipping the CRI detection.
-	// Ref: https://github.com/kubernetes/kubeadm/issues/1559
-	initCfg.NodeRegistration.CRISocket = kubeadmconstants.DefaultDockerCRISocket
-
-	internalcfg, err := configutil.LoadOrDefaultInitConfiguration(cfgPath, initCfg, clusterCfg)
-	if err != nil {
-		return err
-	}
-
+func RunCreateToken(out io.Writer, client clientset.Interface, cfgPath string, bootstrapToken *kubeadmapi.BootstrapTokenString, printJoinCommand bool, kubeConfigFile string) error {
 	klog.V(1).Infoln("[token] creating token")
-	if err := tokenphase.CreateNewTokens(client, internalcfg.BootstrapTokens); err != nil {
+	err := tokenphase.CreateNewTokens(client, []kubeadmapi.BootstrapToken{
+		{
+			Token: bootstrapToken,
```

Comment:
Yes, I will update the PR, because some tests fail when doing so (mainly because validation of bootstrap tokens happen exclusively when read from configuration). My intent is to move the validation to BootstrapToken itself, and let the configuration validation use this logic as well.

Will update the PR with the proposal. Thanks @rosti!